consensus/consensus.go

```diff
@@ -137,6 +139,15 @@ type Engine interface {
 	Close() error
 }
 
+func TargetToDifficulty(target *big.Int) *big.Int {
```

Comment:
Hmm, I don't think this function is guaranteed to be exact. Target may include a rounding error. I know you didn't add this, but are you using it somewhere with the expectation that it yields an exact result?

`target = 2e256/dif` will round down if `dif` is not an even multiple of 2e256.

Reply:
The only places we use this are in the proxy so its not a consensus issue if its not 100% exact.